pygama/dsp/_processors/get_multi_local_extrema.py

```diff
@@ -0,0 +1,93 @@
+import numpy as np
+from numba import guvectorize, njit
+from pygama.dsp.errors import DSPFatal
+
+
+
+@guvectorize(["void(float32[:], float32, float32[:], float32[:], float32[:], float32[:], float32[:], int32[:])",
+              "void(float64[:], float64, float64[:], float64[:], float64[:],float64[:], float64[:], int32[:])",
+              "void(int32[:], int32, int32[:], int32[:], int32[:], int32[:], int32[:], int32[:])",
+              "void(int64[:], int64, int64[:], int64[:], int64[:], int64[:], int64[:], int32[:])"],
+             "(n),(),(m) -> (m),(m),(),(),()", nopython=True, cache=True)
+def get_multi_local_extrema(w_in, delta_in, t_arbitrary_in, vt_max_out, vt_min_out, n_max_out, n_min_out, flag_out):
+    """
+    Get lists of indices of the local maxima and minima of data
+    The "local" extrema are those maxima / minima that have heights / depths of
+    at least delta.
+    Converted from MATLAB script at: http://billauer.co.il/peakdet.html
+    Parameters
+    ----------
+    w_in : array-like
+        The array of data within which extrema will be found
+    delta_in : scalar
+        The absolute level by which data must vary (in one direction) about an
+        extremum in order for it to be tagged
+    t_arbitrary_in : array-like
+        An array of fixed length that tells numba to expect to return a list of amplitudes of the same length
+    Returns
+    -------
+    vt_max_out, vt_min_out : array-like, array-like
+        Arrays of fixed length (padded with nans) that hold the indices of
+        the identified local maxima and minima
+    n_max_out, n_min_out: scalar, scalar 
+        The number of maxima and minima found in a waveform
+    flag_out: scalar
+        Returns 0 if there is only one maximum and it is a simple waveform, 
+        Returns 1 if there are multiple peaks in a waveform
+        
+    """
+
+    # prepare output 
+    #padded with nans or zeros depending on what we want
+    vt_max_out[:]= np.nan
+    vt_min_out[:] = np.nan
+
+    
+    n_max_out[0] = 0
+    n_min_out[0] = 0
+    flag_out[0] = 0
+    
+
+    # sanity checks
+    w_in, delta2 = np.asarray(w_in), np.asarray(delta_in)
+    if delta2.ndim != 0:
+        raise DSPFatal("Input argument delta_in must be a scalar")
```

Comment:
I think that the logic that handles numpy/numba broadcasting should make this check unnecessary, since the value has dimension () in the signature. On the other hand, if you've run into a situation where this has happened, please say so, and in that case we will want to add this check other places!

Reply:
I would not use functions that modify the input waveform. We want all inputs to be treated as const objects. All operations should be done on tmp or output objects

Reply:
I would add the nan checks here: see https://github.com/legend-exp/pygama/blob/5e261e69eec1610431fe2eda42d994ba58d299ea/pygama/dsp/_processors/template.py#L49